qiskit/opflow/evolutions/matrix_evolution.py

```diff
@@ -30,6 +31,13 @@ class MatrixEvolution(EvolutionBase):
     ``UnitaryGates`` or ``HamiltonianGates`` containing the exponentiation of the Operator.
     """
 
+    @deprecate_func(
+        since="0.24.0",
+        additional_msg="This deprecation is part of the opflow module deprecation effort. For code migration guidelines, visit https://qisk.it/opflow_migration.",
+    )
+    def __init__(self) -> None:
+        super().__init__()
```

Comment:
Maybe its mypy rather than lint which tells one this is a rather useless init. I get that its done for the decorator - if it fails CI I guess we just suppress the error.